cogs/MalwareProtection.py

```diff
@@ -0,0 +1,62 @@
+import discord
+from discord.ext import commands
+from .utils import constants
+from os.path import splitext
+from assets import AntiMalware
+
+FILE_EMBED_DESCRIPTION = (
+    "**Uh-oh!** Your message got zapped by our spam filter. "
+    "We currently don't allow `.txt` attahcments or any source files, so here are some tips to help you work safely: \n\n"
+    "• If you attempted to send a message longer than 2000 characters, try shortening your message "
+    "to fit within the character limit or use a pasting service (see below) \n\n"
+    "• If you tried to show someone your code, you can use codeblocks or use a pasting service like: "
+    f"\n{constants.paste_link} or {constants.paste_link_2}"
+)
+
+
+class MalwareProtection(commands.Cog):
+    def __init__(self, client):
+        self.client = client
+
+    @commands.Cog.listener()
+    async def on_message(self, message: discord.Message) -> None:
+        """Find messages with blacklisted attachments."""
+        if not message.attachments or not message.guild:
+            return
+
+        file_extensions = {splitext(attachment.filename.lower())[1] for attachment in message.attachments}
+        is_blocked = file_extensions - set(AntiMalware.whitelist)
+
+        if is_blocked:
+            embed = discord.Embed(color=discord.Color.gold())
+            embed.description = FILE_EMBED_DESCRIPTION
+
+        if embed.description:
+            await message.channel.send(f"Hey {message.author.mention}!", embed=embed)
+
+            try:
+                file_pastes = []
+
+                for attachment in message.attachments:
+                    content = await attachment.read()
+                    async with self.session.post('https://www.hastebin.com/documents', data=content) as resp:
+                        file_paste = 'https://www.hastebin.com/' + (await resp.json())['key']
+
+                        file_pastes.append(file_paste)
+
+                em = discord.Embed(
+                    color=discord.Color.red(),
+                    description=f"The Paste(s) of the File(s) Can be found at {','.join(file_pastes)}",
+                    title="File Pastes!"
+                )
+                await message.delete()
+                await message.channel.send(embed=em)
+            # TODO : Add logging for Everything ASAP.
+            except discord.NotFound:
+                pass
+            except ConnectionError:
+                pass
+
+
+def setup(client):
+    client.add_cog(MalwareProtection(client))
```

Comment:
New standard from #11 requires us to use `bot` not `client` for variable names here
```suggestion
def setup(bot):
    bot.add_cog(MalwareProtection(bot))
```
Also don't forget about **type hinting**